armi/bookkeeping/db/database3.py

```diff
@@ -100,6 +94,168 @@ def getH5GroupName(cycle: int, timeNode: int, statePointName: str = None) -> str
     return "c{:0>2}n{:0>2}{}".format(cycle, timeNode, statePointName or "")
 
 
+class JaggedArray:
```

Comment:
By preference, I would add this new class to it's own file. (This is safe, since `Database3` will import it, but it doesn't import `Database3`.)

The idea is simple: the file `Database3.py` is already 2000 lines long, and has several classes in it. It's not _always_ necessary, of course, but "one file = one class" (when it works) does make it easier to read a code base.

Thoughts?